wrapper_test/src/unit/ut_repo_csharp_interface.cpp

```diff
@@ -37,56 +37,110 @@ static std::string getDataPath(
 	return returnPath;
 }
 
-TEST(RepoCsharpInterface, LoadScene)
+// Only one test for the whole wrapper, as the current implementation is done
+// via the singleton pattern, and the core repo controller state may 
+// be carried over between tests.
+// This test covers all the wrapper functions that are run by the asset bundle 
+// creator via the `BuildAllAssetBundles` entry point
+TEST(RepoCsharpInterface, FullWrapperTest)
 {
-	// using the readly ony database so the project should be in 
-	// the test db, even if a round of tests has already been run on it
-	std::string database = "sampleDataReadOnly";
-	std::string project = "3drepoBIM";
+	// using the read/write section of the test database
+	std::string database = "stUpload";
+	std::string project = "cube";
 	std::string revisionID = "";
 	std::string configPath = getDataPath("config/config.json");
 	bool connected = repoConnect(&configPath[0]);
-	// check connection
 	ASSERT_TRUE(connected);
 	bool loadedScene = repoLoadSceneForAssetBundleGeneration(
 		&database[0],
 		&project[0],
 		&revisionID[0]);
-	// check scene loaded
 	ASSERT_TRUE(loadedScene);
-	// vr enabled check
 	ASSERT_FALSE(repoIsVREnabled());
-	// super mesh count
-	int numSuperMeshes = repoGetNumSuperMeshes();
-	ASSERT_EQ(4, numSuperMeshes);
-	// federation check
 	bool isFederation = repoIsFederation(
 		&database[0],
 		&project[0]);
 	ASSERT_FALSE(isFederation);
-}
-
-TEST(RepoCsharpInterface, AssetBundleSave)
-{
-	// using the read/write database
-	std::string database = "stUpload";
-	std::string project = "cube";
-	std::string revisionID = "";
-	std::string configPath = getDataPath("config/config.json");
-	bool connected = repoConnect(&configPath[0]);
-	// check connection
-	ASSERT_TRUE(connected);
-	// load scene
-	bool loadedScene = repoLoadSceneForAssetBundleGeneration(
-		&database[0],
-		&project[0],
-		&revisionID[0]);
-	// check scene loaded
-	ASSERT_TRUE(loadedScene);
-	// asset bundle save check
+	// main loop to access supermesh data
+	int numSuperMeshes = repoGetNumSuperMeshes();
+	ASSERT_EQ(numSuperMeshes, 1);
+	repoInfo << "looping over supermeshes";
+	for (int superMeshIdx = 0; superMeshIdx < numSuperMeshes; superMeshIdx++)
+	{
+		// get info regarding the current supermesh
+		long numSubSupermeshes{0};
```

Comment:
any reason why we're using the curly brace syntax over just doing a = 0?

Reply:
The curly brace (list initializer) does not allow type narrowing (e.g. float to int) but in this case does not make a difference. I'll change it to the equals syntax.